code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/InputBoxNotificationBuilder.kt

```diff
@@ -0,0 +1,283 @@
+*/
+
+package com.adobe.marketing.mobile.services.ui.notification.builders
+
+import android.app.Activity
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.view.View
+import android.widget.RemoteViews
+import androidx.core.app.NotificationCompat
+import androidx.core.app.RemoteInput
+import com.adobe.marketing.mobile.core.R
+import com.adobe.marketing.mobile.services.Log
+import com.adobe.marketing.mobile.services.ServiceProvider
+import com.adobe.marketing.mobile.services.ui.notification.NotificationConstructionFailedException
+import com.adobe.marketing.mobile.services.ui.notification.PushTemplateConstants
+import com.adobe.marketing.mobile.services.ui.notification.PushTemplateImageUtil
+import com.adobe.marketing.mobile.services.ui.notification.extensions.createNotificationChannelIfRequired
+import com.adobe.marketing.mobile.services.ui.notification.templates.InputBoxPushTemplate
+
+/**
+ * Object responsible for constructing a [NotificationCompat.Builder] object containing an input box push template notification.
+ */
+internal object InputBoxNotificationBuilder {
+    private const val SELF_TAG = "InputBoxNotificationBuilder"
+
+    @Throws(NotificationConstructionFailedException::class)
+    fun construct(
+        context: Context,
+        pushTemplate: InputBoxPushTemplate,
+        trackerActivityClass: Class<out Activity>?,
+        broadcastReceiverClass: Class<out BroadcastReceiver>?
+    ): NotificationCompat.Builder {
+        val cacheService = ServiceProvider.getInstance().cacheService
+            ?: throw NotificationConstructionFailedException("Cache service is null, input box push notification will not be constructed.")
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Building an input box template push notification."
+        )
+        val packageName = context.packageName
+        val smallLayout = RemoteViews(packageName, R.layout.push_template_collapsed)
+        val expandedLayout = RemoteViews(packageName, R.layout.push_template_expanded)
+
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val channelIdToUse: String = notificationManager.createNotificationChannelIfRequired(
+            context,
+            pushTemplate.channelId,
+            pushTemplate.sound,
+            pushTemplate.getNotificationImportance(),
+            pushTemplate.isFromIntent
+        )
+
+        // create the notification builder with the common settings applied
+        val notificationBuilder = AEPPushNotificationBuilder.construct(
+            context,
+            pushTemplate,
+            channelIdToUse,
+            trackerActivityClass,
+            smallLayout,
+            expandedLayout,
+            R.id.basic_expanded_layout
```

Comment:
Why do we reference basic template here?

Reply:
the input box notification uses the basic expanded layout